### PR TITLE
add optional pre-feedback notice

### PR DESCRIPTION
**Pull request by johngruen**
#1328

### CODE REVIEW

File: frontend/src/app/components/ExperimentTourDialog.js
Comment (johngruen) [inline frontend/src/app/components/ExperimentTourDialog.js:101]: I should have taken this out on that refactor of the tour UI

**Comment by johngruen** [comment]
If someone wants to check the UI here, just populate the `pre_feedback_image` and `pre_feedback_copy` fields on one of the experiment YAML files

File: content-src/experiments/activity-stream.yaml
Comment (chuckharmston) [inline content-src/experiments/activity-stream.yaml:21]: We should probably build the front-end in such a way that we don't need to include empty keys at all.

File: frontend/src/app/components/ExperimentPage.js
Comment (chuckharmston) [inline frontend/src/app/components/ExperimentPage.js:40]: This should probably be in the redux store, per #1367.

File: frontend/src/styles/modules/_modals.scss
Comment (chuckharmston) [inline frontend/src/styles/modules/_modals.scss]: Throw parens around this? Way easier to read.

File: locales/en-US/app.ftl
Comment (chuckharmston) [inline locales/en-US/app.ftl]: Needs a space after `=`

File: frontend/src/app/components/ExperimentPreFeedbackDialog.js
Comment (chuckharmston) [inline frontend/src/app/components/ExperimentPreFeedbackDialog.js:29]: I would really rather not do this. What's the use case?

**Review by chuckharmston — CHANGES_REQUESTED** [review]

File: frontend/src/app/components/ExperimentPage.js
Comment (johngruen) [reply frontend/src/app/components/ExperimentPage.js:40]: @chuckharmston so, should i do anything about this now, or are you just noting for the future?

Comment (chuckharmston) [reply frontend/src/app/components/ExperimentPage.js:40]: I'd rather you fix it now, but I won't block merge for it.

File: content-src/experiments/activity-stream.yaml
Comment (johngruen) [reply content-src/experiments/activity-stream.yaml:21]: yeah, we don't need to...I just figured we should keep the various files uniform

File: frontend/src/app/components/ExperimentPage.js
Comment (johngruen) [reply frontend/src/app/components/ExperimentPage.js:40]: Let's do all of this in #1367, since we're going to be adding all of this stuff to the redux store there anyway.

File: frontend/src/app/components/ExperimentPreFeedbackDialog.js
Comment (johngruen) [reply frontend/src/app/components/ExperimentPreFeedbackDialog.js:29]: @chuckharmston yeah, not necessary here

**Comment by johngruen** [comment]
Alright, should be resolved % redux store stuff, which should be done collectively in #1367

File: content-src/experiments/activity-stream.yaml
Comment (johngruen) [reply content-src/experiments/activity-stream.yaml:21]: I mean, these fields don;t need to be here...i just figured we could keep the yaml files consistent

File: frontend/src/app/components/ExperimentPreFeedbackDialog.js
Comment (johngruen) [reply frontend/src/app/components/ExperimentPreFeedbackDialog.js:29]: @chuckharmston @lmorchard actually, it appears to be, though i'm not exactly sure why...but I'm seeing the same thing elsewhere (ie. in the ExperimentTourDialogue)

File: frontend/src/app/components/ExperimentPage.js
Comment (johngruen) [inline frontend/src/app/components/ExperimentPage.js]: @chuckharmston added the 'undefined' thing to the test here if the pre_feedback fields are not in the model

**Comment by lmorchard** [comment]
Looks like changes make this ready for another review? Will take a peek.

File: content-src/experiments/activity-stream.yaml
Comment (lmorchard) [reply content-src/experiments/activity-stream.yaml:21]: This could be a job for #1308 - can leave things out / decide what's required and what's valid in a linting step. We should, like, document what's in these YAML files someday soon too

File: frontend/src/app/components/ExperimentPage.js
Comment (lmorchard) [reply frontend/src/app/components/ExperimentPage.js:40]: It's getting kinda hectic, so we can borrow some tech-debt and make #1367 bigger IMO

File: frontend/src/app/components/ExperimentPage.js
Comment (lmorchard) [reply frontend/src/app/components/ExperimentPage.js]: Should this be `typeof(pre_feedback_copy) === 'undefined'`? You could probably get away with just a `!pre_feedback_copy` unless we ever want the copy to literally be "0". Also should be a full if / else statement rather than a ternary condition, since we're not doing anything with the result value.

File: frontend/src/app/components/ExperimentPreFeedbackDialog.js
Comment (lmorchard) [reply frontend/src/app/components/ExperimentPreFeedbackDialog.js:29]: Assuming we want markup allowed in `pre_feedback_copy`, [this is the way to do it](https://facebook.github.io/react/tips/dangerously-set-inner-html.html) as far as I understand. It looks ugly as a warning. But, I think it's okay because it's from our own JSON & YAML and we don't need to sanitize our own content.

**Review by lmorchard — COMMENTED** [review]
Only really one nit, with the ternary/if thing. Adding a little bit for #1367 to do, but that's probably okay for now? Maybe we need a //TODO comment there so we can tally our debt someday?

**Comment by johngruen** [comment]
@chuckharmston @lmorchard all requested changes made

**Review by lmorchard — APPROVED** [review]
Works for me

**Comment by lmorchard** [comment]
Not sure how to unblock this for a merge, though. My review approves, but doesn't seem to override the one still requesting changes

**Review by chuckharmston — APPROVED** [review]

**Comment by chuckharmston** [comment]
Approving and merging based on Les' r+

**Pull request merged**
